packages/skeleton/src/components/project.ts

```diff
@@ -0,0 +1,462 @@
+import fs from 'fs'
+import path from 'path'
+
+import chalk from 'chalk'
+
+import { getPaths } from '@redwoodjs/internal/dist/index'
+
+import { getRootPath } from '../lib/path'
+
+import { extractCells } from './cell'
+import type { RedwoodCell } from './cell'
+import { RedwoodError, RedwoodWarning } from './diagnostic'
+import { extractDirectives } from './directive'
+import type { RedwoodDirective } from './directive'
+import { extractFunctions } from './function'
+import type { RedwoodFunction } from './function'
+import { extractLayouts } from './layout'
+import type { RedwoodLayout } from './layout'
+import { extractPages } from './page'
+import type { RedwoodPage } from './page'
+import { extractRouters } from './router'
+import type { RedwoodRouter } from './router'
+import { extractSDLs } from './sdl/sdl'
+import type { RedwoodSDL } from './sdl/sdl'
+import { extractServices } from './service/service'
+import type { RedwoodService } from './service/service'
+import { extractSides } from './side'
+import type { RedwoodSide } from './side'
+import { RedwoodSkeleton } from './skeleton'
+import { extractTOMLs } from './toml'
+import type { RedwoodTOML } from './toml'
+
+/**
+ * Used to enumerate either JS or TS project types
+ */
+export enum RedwoodProjectType {
+  TYPESCRIPT = 'typescript',
+  JAVASCRIPT = 'javascript',
+}
+
+export class RedwoodProject extends RedwoodSkeleton {
+  warnings: RedwoodWarning[] = []
+  errors: RedwoodError[] = []
```

Comment:
These aren't marked as `abstract` in `RedwoodSkeleton`, so this strikes me as redundant: 

```suggestion
```